### PR TITLE
`azurerm_kubernetes_cluster` - Fix error when update `outbound_type` in `network_profile` nested block

### LINKED ISSUE #16712: azurerm_kubernetes_cluster - AKS cluster creation fails when load_balancer_profile block not empty and outboundType=userAssignedNATGateway
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.7

### AzureRM Provider Version

2.99

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "aks" {
  depends_on              = [azurerm_subnet_nat_gateway_association.aks_nat_gw]
  name                    = var.prefix
  location                = azurerm_resource_group.rg.location
  resource_group_name     = azurerm_resource_group.rg.name
  node_resource_group     = "${var.prefix}-nodes-rg"
  dns_prefix              = var.prefix
  sku_tier                = "Paid"
  private_cluster_enabled = false

  kubernetes_version = data.azurerm_kubernetes_service_versions.current.latest_version

  automatic_channel_upgrade = "patch" 

  auto_scaler_profile {
    balance_similar_node_groups      = true
    max_graceful_termination_sec     = "600"
    scale_down_utilization_threshold = "0.7"
    skip_nodes_with_local_storage    = false # Nodes should use local storage only as cache or temporary not as persistent storage
    skip_nodes_with_system_pods      = false # Pods in kube-system won't block the deployment and must be HA
  }

  addon_profile {
    http_application_routing {
      enabled = false
    }
    kube_dashboard {
      enabled = false
    }
    azure_policy {
      enabled = true
    }
  }

  default_node_pool {
    name                         = "system"
    os_disk_size_gb              = var.system_pool_config.disk_size
    os_disk_type                 = var.system_pool_config.os_disk_type
    vm_size                      = var.system_pool_config.vm_size
    min_count                    = var.system_pool_config.min_count
    max_count                    = var.system_pool_config.max_count
    vnet_subnet_id               = azurerm_subnet.subnet.id
    tags                         = var.tags
    only_critical_addons_enabled = true
    enable_auto_scaling          = true
    enable_host_encryption       = true
    max_pods                     = 30 
    availability_zones           = [1, 2, 3]
    orchestrator_version         = var.system_pool_config.orchestrator_version

    upgrade_settings {
      max_surge = var.system_pool_config.max_surge
    }

  }
  identity {
    type = "SystemAssigned"
  }

  linux_profile {
    admin_username = "kubernetes"
    ssh_key {
      key_data = tls_private_key.node-ssh-key.public_key_openssh
    }
  }

  role_based_access_control {
    enabled = true

    azure_active_directory {
      managed                = "true"
      admin_group_object_ids = var.admin_group_object_ids
    }
  }


  network_profile {
    network_plugin     = "azure"
    network_mode       = "transparent"
    network_policy     = "calico"
    service_cidr       = "10.0.0.0/16"
    dns_service_ip     = "10.0.0.10"
    docker_bridge_cidr = "172.17.0.1/16"
    load_balancer_sku  = "standard"

    load_balancer_profile {
      outbound_ports_allocated  = 0
      idle_timeout_in_minutes   = 4
      managed_outbound_ip_count = 1
    }
    outbound_type = "userAssignedNATGateway"
  }

  tags = var.tags
}
```

### Debug Output/Panic Output

```shell
waiting for creation of Cluster: (Managed Cluster Name "managed-cluster" / Resource Group "managed-cluster-rg"): Code="CreateVMSSAgentPoolFailed" Message="AKS encountered an internal error while attempting the requested Creating operation. AKS will continuously retry the requested operation until successful or a retry timeout is hit. Check back to see if the operation requires resubmission.
```

### Expected Behaviour

The managed cluster is deployed

### Actual Behaviour

Its always failing with the error message above. 
Also ported the to AzureRM 3.5 but the error is the same.

Strange thing, before 1st of May it worked without any problems. After 1st of May it started failing with the error of
`Failure sending request: StatusCode=400 -- Original Error: Code="SubnetsAssociatedWithNATgatewayWhenOutboundTypeIsStandardLoadBalancer"`
thatswhy the `outbound_type = "userAssignedNATGateway"` introduced

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

### LINKED ISSUE #25499: Changing `outboundType`  to `userDefinedRouting` leads malfunction error
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

1. Deploy AKS cluster with outbound load balancer setting, without route table and firewall.

2. Change `outboundType` to `userDefinedRouting`
```tf
resource "azurerm_kubernetes_cluster" "k8s" {
  count               = var.resource_count
  name                = local.cluster_names[count.index]
  resource_group_name = azurerm_resource_group.k8s-rg[count.index].name
  location            = azurerm_resource_group.k8s-rg[count.index].location

  dns_prefix = var.dns_prefix
  kubernetes_version = "1.27.7"

  default_node_pool {
    name = "default"
    node_count = var.agent_count
    vm_size = "Standard_D2_v2"
    vnet_subnet_id = azurerm_subnet.node-subnet[count.index].id
    pod_subnet_id = azurerm_subnet.pod-subnet[count.index].id
  }

  identity {
    type = "SystemAssigned"
  }

  ingress_application_gateway {
    gateway_id = azurerm_application_gateway.network[count.index].id
  }

  network_profile {
    load_balancer_sku = "standard"
    network_plugin = "azure"
    outbound_type = "loadBalancer" => "userDefinedRouting"
  }
  tags = {
    Environment = "Development"
    delegation = "trial-01"
  }
}

```
3. Error happened even not specifying `loadBalacerProfile`
```bash
Error: updating Kubernetes Cluster (Subscription: "43fec997-2a49-4f32-8cec-81e390065f4f"

│ Resource Group Name: "aks-rg-0"
│ Kubernetes Cluster Name: "aks-0"): performing CreateOrUpdate: unexpected status 400 with response: {
│   "code": "InvalidUserDefinedRoutingWithLoadBalancerProfile",
│   "details": null,
│   "message": "UserDefinedRouting and load balancer profile are mutually exclusive. Please refer to [http://aka.ms/aks/outboundtype](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Faka.ms%2Faks%2Foutboundtype&data=05%7C02%7Candrewmin%40microsoft.com%7C89642caae95a4a25cad608dc53a3a333%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638477209564450897%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=1cV53KRYKBWkFelvZE%2Bv2vcgOq56qv4Z6vIbjCAsavs%3D&reserved=0) for more details",
│   "subcode": "",
│   "target": "networkProfile.loadBalancerProfile"
│  }
│
│   with azurerm_kubernetes_cluster.k8s[0],
│   on aks.tf line 1, in resource "azurerm_kubernetes_cluster" "k8s":
│    1: resource "azurerm_kubernetes_cluster" "k8s" {
│
```

cc: @lonegunmanb 

### Terraform Version

1.0

### AzureRM Provider Version

3.82

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
Refer to https://github.com/n-y-kim/AKS-tf-error-test/blob/main/aks.tf
```

### Debug Output/Panic Output

```shell
described above
```

### Expected Behaviour

working changing `outboundType` to `userDefinedRouting`

### Actual Behaviour

error populating

UserDefinedRouting and load balancer profile are mutually exclusive.

### Steps to Reproduce

Described above

### Important Factoids

_No response_

### References

https://github.com/n-y-kim/AKS-tf-error-test/blob/main/aks.tf

### PR DESCRIPTION
**Pull request by lonegunmanb**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

This pr fixed the error when trying to update `outbound_type` from `loadBalancer` to `userDefinedRouting` or `userAssignedNATGateway`. The original implementation won't erase load balance profile when the `outbound_type` is no longer `loadBalancer`, this mixed-state would be considered as invalid config by the service side. This pr will set the corresponding sub-profile to `nil` when the `outbound_type` has been changed. It should fix issue #25499 and #16712.

## PR Checklist

- [ ] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [ ] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [ ] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [ ] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

A new test has been added.

=== RUN   TestAccKubernetesCluster_updateNetworkProfileOutboundType
=== PAUSE TestAccKubernetesCluster_updateNetworkProfileOutboundType
=== CONT  TestAccKubernetesCluster_updateNetworkProfileOutboundType
--- PASS: TestAccKubernetesCluster_updateNetworkProfileOutboundType (2786.31s)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [*] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #25499, fixes #16712

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Comment by lonegunmanb** [comment]
I just read pr #20328 submitted by @stephybun , I'd like to merge this pr and fix the issue, but according to your comment: 

>since it appears some load balancer configuration is applicable when outbound type is set to NAT.

Is there anything I can to make this pr been merged? Which part of load balancer configuration should be preserved?

**Comment by lonegunmanb** [comment]
A kindly ping @stephybun is there any chance this pr get merged? Thanks!

**Review by katbyte — APPROVED** [review]
LGTM 🌮

File: internal/services/containers/kubernetes_cluster_resource_test.go [hunk 1]
```diff
@@ -202,6 +202,34 @@ func TestAccKubernetesCluster_edgeZone(t *testing.T) {
 	})
 }
 
+func TestAccKubernetesCluster_updateNetworkProfileOutboundType(t *testing.T) {
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_resource_test.go:205]: Would you mind removing the test `TestAccKubernetesCluster_outboundTypeUpdate` which also tests the update of the outbound type but is less comprehensive.

**Comment by katbyte** [comment]
@lonegunmanb - this PR is good to merge once you address stephs comment above

**Comment by lonegunmanb** [comment]
@stephybun @katbyte apology for the late reply & updated, would you please give this pr another review? Thanks!

**Review by stephybun — APPROVED** [review]
Thanks @lonegunmanb LGTM 👍

**Pull request merged**